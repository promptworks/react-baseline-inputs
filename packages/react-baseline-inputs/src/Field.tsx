import * as React from "react";

export interface FieldProps {
  name: string;
  label: React.ReactNode;
  render: (props: object) => React.ReactNode;

  id?: string;
  inline?: boolean;
  large?: boolean;
  small?: boolean;
  touched?: boolean;
  success?: boolean;
  error?: React.ReactNode;
  help?: React.ReactNode;

  fieldClassName?: string;
  fieldInlineClassName?: string;
  fieldValidClassName?: string;
  fieldInvalidClassName?: string;
  fieldTouchedClassName?: string;
  fieldLargeClassName?: string;
  fieldSmallClassName?: string;

  inputClassName?: string;
  inputValidClassName?: string;
  inputInvalidClassName?: string;
  inputTouchedClassName?: string;
  inputInlineClassName?: string;
  inputLargeClassName?: string;
  inputSmallClassName?: string;

  labelClassName?: string;
  labelInlineClassName?: string;
  labelLargeClassName?: string;
  labelSmallClassName?: string;

  helpClassName?: string;
  errorClassName?: string;
}

const generateUniqueId = (() => {
  let previousId = 0;
  return (name: string) => {
    const id = ++previousId;
    return id === 1 ? name : `${name}_${id}`;
  };
})();

export const Field: React.FC<FieldProps> = ({
  name,
  label,
  render,
  id = React.useMemo(() => generateUniqueId(name), []),

  help,
  inline,
  touched,
  error,
  success = touched && !error,
  large,
  small,

  fieldClassName = "field",
  fieldInlineClassName = "field--inline",
  fieldInvalidClassName = "field--erroneous",
  fieldTouchedClassName = "field--touched",
  fieldValidClassName = "field--success",
  fieldLargeClassName = "field--large",
  fieldSmallClassName = "field--small",

  inputClassName = "field__input",
  inputInlineClassName = "field__input--inline",
  inputInvalidClassName = "field__input--erroneous",
  inputTouchedClassName = "field__input--touched",
  inputValidClassName = "field__input--success",
  inputLargeClassName = "field__input--large",
  inputSmallClassName = "field__input--small",

  labelClassName = "field__label",
  labelInlineClassName = "field__label--inline",
  labelLargeClassName = "field__label--large",
  labelSmallClassName = "field__label--small",

  helpClassName = "field__help",
  errorClassName = "message message--problem"
}) => {
  const fieldClassNames = [fieldClassName];
  const inputClassNames = [inputClassName];
  const labelClassNames = [labelClassName];

  if (inline) {
    fieldClassNames.push(fieldInlineClassName);
    inputClassNames.push(inputInlineClassName);
    labelClassNames.push(labelInlineClassName);
  }

  if (touched) {
    fieldClassNames.push(fieldTouchedClassName);
    inputClassNames.push(inputTouchedClassName);
  }

  if (error) {
    fieldClassNames.push(fieldInvalidClassName);
    inputClassNames.push(inputInvalidClassName);
  }

  if (success) {
    fieldClassNames.push(fieldValidClassName);
    inputClassNames.push(inputValidClassName);
  }

  if (large) {
    fieldClassNames.push(fieldLargeClassName);
    inputClassNames.push(inputLargeClassName);
    labelClassNames.push(labelLargeClassName);
  }

  if (small) {
    fieldClassNames.push(fieldSmallClassName);
    inputClassNames.push(inputSmallClassName);
    labelClassNames.push(labelSmallClassName);
  }

  const inputProps = {
    id,
    name,
    className: inputClassNames.join(" ")
  };

  return (
    <div className={fieldClassNames.join(" ")}>
      {inline && render(inputProps)}

      {label && (
        <label className={labelClassNames.join(" ")} htmlFor={id}>
          {label}
          {help && <span className={helpClassName}>{help}</span>}
        </label>
      )}

      {!inline && render(inputProps)}

      {error && (
        <span role="alert" className={errorClassName}>
          {error}
        </span>
      )}
    </div>
  );
};
