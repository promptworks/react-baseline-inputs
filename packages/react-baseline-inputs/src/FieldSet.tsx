import * as React from "react";
import { FieldSetProps, FieldSetTheme } from "./types";
import { useTheme } from "./theme";
import { generateUniqueId, join } from "./utils";

export const FieldSet: React.FC<FieldSetProps> = ({
  legend,
  id = React.useMemo(generateUniqueId, []),
  help,
  children,
  className,
  theme = useTheme("fieldSet"),
  touched,
  error,
  success = touched && !error,
  large,
  small
}) => {
  const fieldSetTheme = theme as FieldSetTheme;
  const fieldSetClassNames = [fieldSetTheme.fieldSet];
  const legendClassNames = [fieldSetTheme.legend];
  const helpClassNames = [fieldSetTheme.help];
  const errorClassNames = [fieldSetTheme.error];

  if (className) {
    fieldSetClassNames.push(className);
  }

  if (error) {
    fieldSetClassNames.push(fieldSetTheme.fieldSetError);
    legendClassNames.push(fieldSetTheme.legendError);
  }

  if (success) {
    fieldSetClassNames.push(fieldSetTheme.fieldSetSuccess);
    legendClassNames.push(fieldSetTheme.legendSuccess);
  }

  if (touched) {
    fieldSetClassNames.push(fieldSetTheme.fieldSetTouched);
    legendClassNames.push(fieldSetTheme.legendTouched);
  }

  if (large) {
    fieldSetClassNames.push(fieldSetTheme.fieldSetLarge);
    legendClassNames.push(fieldSetTheme.legendLarge);
    errorClassNames.push(fieldSetTheme.errorLarge);
    helpClassNames.push(fieldSetTheme.helpLarge);
  }

  if (small) {
    fieldSetClassNames.push(fieldSetTheme.fieldSetSmall);
    legendClassNames.push(fieldSetTheme.legendSmall);
    errorClassNames.push(fieldSetTheme.errorSmall);
    helpClassNames.push(fieldSetTheme.helpSmall);
  }

  const errorLabelId = `${id}_error`;

  return (
    <fieldset
      id={id}
      aria-describedby={error ? errorLabelId : undefined}
      className={join(fieldSetClassNames)}
    >
      <legend className={join(legendClassNames)}>{legend}</legend>

      {help && <p className={join(helpClassNames)}>{help}</p>}

      {children}

      {error && (
        <span role="alert" className={join(errorClassNames)} id={errorLabelId}>
          {error}
        </span>
      )}
    </fieldset>
  );
};
