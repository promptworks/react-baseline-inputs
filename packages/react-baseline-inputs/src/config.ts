import { Config } from "./types";
import { useContext, createContext } from "react";

export const defaultTheme = {
  field: "field",
  fieldInline: "field--inline",
  fieldSuccess: "field--success",
  fieldError: "field--erroneous",
  fieldTouched: "field--touched",
  fieldLarge: "field--large",
  fieldSmall: "field--small",
  fieldDisabled: "field--disabled",

  input: "field__input",
  inputInline: "field__input--inline",
  inputSuccess: "field__input--success",
  inputError: "field__input--erroneous",
  inputTouched: "field__input--touched",
  inputLarge: "field__input--large",
  inputSmall: "field__input--small",
  inputDisabled: "field__input--disabled",

  label: "field__label",
  labelInline: "field__label--inline",
  labelSuccess: "field__label--success",
  labelError: "field__label--erroneous",
  labelTouched: "field__label--touched",
  labelLarge: "field__label--large",
  labelSmall: "field__label--small",
  labelDisabled: "field__label--disabled",

  help: "field__help",
  helpInline: "field__help--inline",
  helpSmall: "field__help--small",
  helpLarge: "field__help--large",

  error: "field__error",
  errorInline: "field__error--inline",
  errorSmall: "field__error--small",
  errorLarge: "field__error--large"
};

export const defaultConfig: Config = {
  input: defaultTheme,
  checkbox: defaultTheme,
  dateInput: defaultTheme,
  dateTimeInput: defaultTheme,
  field: defaultTheme,
  fileInput: defaultTheme,
  floatInput: defaultTheme,
  integerInput: defaultTheme,
  radioGroup: defaultTheme,
  select: defaultTheme,
  textarea: defaultTheme,
  toggleButton: defaultTheme
};

export const ThemeContext = createContext(defaultConfig);
export const ThemeProvider = ThemeContext.Provider;

export const useTheme = (element: keyof Config) => {
  const config = useContext(ThemeContext);
  return config[element];
};
