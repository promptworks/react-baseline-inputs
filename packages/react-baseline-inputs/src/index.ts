export * from "./Checkbox";
export * from "./Field";
export * from "./DateInput";
export * from "./DateTimeInput";
export * from "./FileInput";
export * from "./FloatInput";
export * from "./IntegerInput";
export * from "./Select";
export * from "./TextArea";
export * from "./TextInput";
export * from "./types";
