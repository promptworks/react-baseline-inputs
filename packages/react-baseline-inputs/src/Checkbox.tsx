import * as React from "react";
import { CustomInputProps } from "./types";
import { Field } from "./Field";

export type CheckboxProps = CustomInputProps<"input", boolean | null>;

/**
 * An HTML `<input />`, but with the following benefits:
 *
 *   * It accepts `null` as a value.
 *   * It emits a `null` value to the `onChange` handler when the input is empty.
 */
export const Checkbox: React.FC<CheckboxProps> = ({
  onChange,
  value,
  label = false,
  wrap = true,
  ...props
}) => {
  const handleChange = React.useCallback(
    event => onChange(event.target.checked),
    [onChange]
  );

  return (
    <Field
      label={label}
      render={inputProps => (
        <input
          {...inputProps}
          type="checkbox"
          checked={Boolean(value)}
          onChange={handleChange}
        />
      )}
      {...props}
    />
  );
};
