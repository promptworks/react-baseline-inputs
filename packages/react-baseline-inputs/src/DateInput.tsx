import * as React from "react";
import { Field } from "./Field";
import { DateInputProps } from "./types";
import { useTheme } from "./theme";

const parse = (value: string): string | null => {
  const date = new Date(value);
  const isValid = !isNaN(date.getTime());
  return isValid ? value : null;
};

/**
 * An HTML `<input type="date" />`, but with the following benefits:
 *
 *   * It accepts `null` as a value.
 *   * It emits an ISO-formatted date string when changed.
 */
export const DateInput: React.FC<DateInputProps> = ({
  value,
  onChange,
  theme = useTheme("dateInput"),
  ...props
}) => {
  const handleChange = React.useCallback(
    event => onChange(parse(event.target.value)),
    [onChange]
  );

  return (
    <Field
      theme={theme}
      render={inputProps => (
        <input
          type="date"
          value={value || ""}
          onChange={handleChange}
          {...inputProps}
        />
      )}
      {...props}
    />
  );
};
