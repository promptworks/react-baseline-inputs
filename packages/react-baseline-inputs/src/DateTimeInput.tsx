import * as React from "react";
import { CustomInputProps } from "./types";
import format from "date-fns/format";
import parse from "date-fns/parse";
import isValid from "date-fns/is_valid";
import { Field } from "./Field";

export type DateTimeInputProps = CustomInputProps<"input", string | null>;

/**
 * An HTML `<input type="datetime-local" />`, but with the following benefits:
 *
 *   * It accepts `null` as a value.
 *   * It emits an ISO-formatted datetime string when changed.
 */
export const DateTimeInput: React.FC<DateTimeInputProps> = ({
  value,
  onChange,
  label = false,
  wrap = true,
  ...props
}) => {
  const formattedValue = React.useMemo(
    () => (value ? format(value, "YYYY-MM-DDTHH:mm:ss") : ""),
    [value]
  );

  const handleChange = React.useCallback(
    event => {
      const value = parse(event.target.value);

      if (isValid(value)) {
        onChange(value.toISOString());
      } else {
        onChange(null);
      }
    },
    [onChange]
  );

  return (
    <Field
      label={label}
      render={inputProps => (
        <input
          {...inputProps}
          type="datetime-local"
          value={formattedValue}
          onChange={handleChange}
        />
      )}
      {...props}
    />
  );
};
