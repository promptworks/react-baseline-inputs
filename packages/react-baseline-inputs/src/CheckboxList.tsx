import * as React from "react";
import { Field } from "./Field";
import { FieldSet } from "./FieldSet";
import { CheckboxListProps, OptionProps } from "./types";
import { useTheme } from "./theme";

const getCheckboxProps = (option: OptionProps | string) => {
  if (typeof option === "string") {
    return { value: option, key: option, label: option };
  }

  const { value, label = value, key = value, ...props } = option;
  return { value, label, key, ...props };
};

const remove = (items: any[], item: any) => {
  const result = items.concat([]);
  result.splice(items.indexOf(item), 1);
  return result;
};

export const CheckboxList: React.FC<CheckboxListProps> = ({
  value,
  onChange,
  options = [],
  theme = useTheme("checkbox"),
  title,
  wrapper = true,
  error,
  touched,
  ...props
}) => {
  const values = value || [];
  const handleChange = React.useCallback(
    event => {
      const { checked, value } = event.target;

      if (checked) {
        onChange([...values, value]);
      } else {
        onChange(remove(values, value));
      }
    },
    [onChange, value]
  );

  const [Wrapper, wrapperProps] = wrapper
    ? [FieldSet, { legend: title, error, touched }]
    : [React.Fragment, {}];

  return (
    <Wrapper {...wrapperProps}>
      {options.map(option => {
        const checkbox = getCheckboxProps(option);

        return (
          <Field
            theme={theme}
            key={checkbox.key}
            label={checkbox.label}
            labelPosition="after"
            disabled={checkbox.disabled}
            render={inputProps => (
              <input
                type="checkbox"
                checked={values.includes(checkbox.value)}
                value={checkbox.value}
                onChange={handleChange}
                {...inputProps}
              />
            )}
            {...props}
          />
        );
      })}
    </Wrapper>
  );
};
