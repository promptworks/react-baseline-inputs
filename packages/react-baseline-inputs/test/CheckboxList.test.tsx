import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import { itBehavesLikeAField } from "./sharedExamples";
import { CheckboxList, CheckboxListProps } from "../src";

const setup = (props: Partial<CheckboxListProps> = {}) =>
  render(
    <CheckboxList
      options={[
        { label: "Foo", value: "foo" },
        { label: "Bar", value: "bar" },
        { label: "Disabled", value: "buzz", disabled: true }
      ]}
      value={null}
      onChange={jest.fn()}
      {...props}
    />
  );

describe("<CheckboxList />", () => {
  itBehavesLikeAField(setup, ["label", "wrapper", "id"]);

  test("adds a new value when checked", () => {
    const onChange = jest.fn();
    const { getByLabelText } = setup({ onChange });

    fireEvent.click(getByLabelText("Foo"));
    expect(onChange).toHaveBeenCalledWith(["foo"]);
  });

  test("removes an existing value when unchecked", () => {
    const onChange = jest.fn();
    const { getByLabelText } = setup({ onChange, value: ["foo"] });

    fireEvent.click(getByLabelText("Foo"));
    expect(onChange).toHaveBeenCalledWith([]);
  });

  describe("an array of strings as options", () => {
    const options = ["foo", "bar", "buzz"];

    it("accepts an array of strings as options", () => {
      const { container } = setup({ options });
      expect(container.firstChild).toMatchSnapshot();
    });

    it("adds a new value when checked", () => {
      const onChange = jest.fn();
      const { getByLabelText } = setup({ options, onChange });

      fireEvent.click(getByLabelText("foo"));
      expect(onChange).toHaveBeenCalledWith(["foo"]);
    });

    test("removes an existing value when unchecked", () => {
      const onChange = jest.fn();
      const { getByLabelText } = setup({ options, onChange, value: ["foo"] });

      fireEvent.click(getByLabelText("foo"));
      expect(onChange).toHaveBeenCalledWith([]);
    });
  });

  describe("<fieldset>", () => {
    const options = ["foo", "bar", "buzz"];

    it("is wrapped in a <fieldset> by default", () => {
      const { container } = setup({ options });
      expect(container.firstChild).toMatchSnapshot();
    });

    it("can be rendered unwrapped", () => {
      const { container } = setup({ options, wrapper: false });
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
