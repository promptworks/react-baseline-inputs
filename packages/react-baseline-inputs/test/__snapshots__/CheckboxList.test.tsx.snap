// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxList /> <fieldset> can be rendered unwrapped 1`] = `
<div
  class="field"
>
  <input
    class="field__input"
    id="field_57"
    type="checkbox"
    value="foo"
  />
  <label
    class="field__label"
    for="field_57"
  >
    foo
  </label>
</div>
`;

exports[`<CheckboxList /> <fieldset> is wrapped in a <fieldset> by default 1`] = `
<fieldset
  class="fieldset"
  id="field_53"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_54"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label"
      for="field_54"
    >
      foo
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_55"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label"
      for="field_55"
    >
      bar
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_56"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label"
      for="field_56"
    >
      buzz
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> an array of strings as options accepts an array of strings as options 1`] = `
<fieldset
  class="fieldset"
  id="field_41"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_42"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label"
      for="field_42"
    >
      foo
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_43"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label"
      for="field_43"
    >
      bar
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_44"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label"
      for="field_44"
    >
      buzz
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> renders 1`] = `
<fieldset
  class="fieldset"
  id="field_1"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_2"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label"
      for="field_2"
    >
      Foo
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_3"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label"
      for="field_3"
    >
      Bar
    </label>
  </div>
  <div
    class="field field--disabled"
  >
    <input
      class="field__input field__input--disabled"
      disabled=""
      id="field_4"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--disabled"
      for="field_4"
    >
      Disabled
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> renders with an error 1`] = `
<fieldset
  aria-describedby="field_5_error"
  class="fieldset fieldset--erroneous"
  id="field_5"
>
  <legend
    class="fieldset__legend fieldset__legend--erroneous"
  />
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_6"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label"
      for="field_6"
    >
      Foo
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_7"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label"
      for="field_7"
    >
      Bar
    </label>
  </div>
  <div
    class="field field--disabled"
  >
    <input
      class="field__input field__input--disabled"
      disabled=""
      id="field_8"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--disabled"
      for="field_8"
    >
      Disabled
    </label>
  </div>
  <span
    class="fieldset__error"
    id="field_5_error"
    role="alert"
  >
    Oh no!
  </span>
</fieldset>
`;

exports[`<CheckboxList /> renders with help 1`] = `
<fieldset
  class="fieldset"
  id="field_9"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_10"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label"
      for="field_10"
    >
      Foo
      <span
        class="field__help"
      >
        Good luck
      </span>
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_11"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label"
      for="field_11"
    >
      Bar
      <span
        class="field__help"
      >
        Good luck
      </span>
    </label>
  </div>
  <div
    class="field field--disabled"
  >
    <input
      class="field__input field__input--disabled"
      disabled=""
      id="field_12"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--disabled"
      for="field_12"
    >
      Disabled
      <span
        class="field__help"
      >
        Good luck
      </span>
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> renders with inline 1`] = `
<fieldset
  class="fieldset"
  id="field_29"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field field--inline"
  >
    <input
      class="field__input field__input--inline"
      id="field_30"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label field__label--inline"
      for="field_30"
    >
      Foo
    </label>
  </div>
  <div
    class="field field--inline"
  >
    <input
      class="field__input field__input--inline"
      id="field_31"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label field__label--inline"
      for="field_31"
    >
      Bar
    </label>
  </div>
  <div
    class="field field--inline field--disabled"
  >
    <input
      class="field__input field__input--inline field__input--disabled"
      disabled=""
      id="field_32"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--inline field__label--disabled"
      for="field_32"
    >
      Disabled
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> renders with large 1`] = `
<fieldset
  class="fieldset"
  id="field_21"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field field--large"
  >
    <input
      class="field__input field__input--large"
      id="field_22"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label field__label--large"
      for="field_22"
    >
      Foo
    </label>
  </div>
  <div
    class="field field--large"
  >
    <input
      class="field__input field__input--large"
      id="field_23"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label field__label--large"
      for="field_23"
    >
      Bar
    </label>
  </div>
  <div
    class="field field--large field--disabled"
  >
    <input
      class="field__input field__input--large field__input--disabled"
      disabled=""
      id="field_24"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--large field__label--disabled"
      for="field_24"
    >
      Disabled
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> renders with small 1`] = `
<fieldset
  class="fieldset"
  id="field_25"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field field--small"
  >
    <input
      class="field__input field__input--small"
      id="field_26"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label field__label--small"
      for="field_26"
    >
      Foo
    </label>
  </div>
  <div
    class="field field--small"
  >
    <input
      class="field__input field__input--small"
      id="field_27"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label field__label--small"
      for="field_27"
    >
      Bar
    </label>
  </div>
  <div
    class="field field--small field--disabled"
  >
    <input
      class="field__input field__input--small field__input--disabled"
      disabled=""
      id="field_28"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--small field__label--disabled"
      for="field_28"
    >
      Disabled
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> renders with success 1`] = `
<fieldset
  class="fieldset"
  id="field_13"
>
  <legend
    class="fieldset__legend"
  />
  <div
    class="field field--success"
  >
    <input
      class="field__input field__input--success"
      id="field_14"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label field__label--success"
      for="field_14"
    >
      Foo
    </label>
  </div>
  <div
    class="field field--success"
  >
    <input
      class="field__input field__input--success"
      id="field_15"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label field__label--success"
      for="field_15"
    >
      Bar
    </label>
  </div>
  <div
    class="field field--success field--disabled"
  >
    <input
      class="field__input field__input--success field__input--disabled"
      disabled=""
      id="field_16"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--success field__label--disabled"
      for="field_16"
    >
      Disabled
    </label>
  </div>
</fieldset>
`;

exports[`<CheckboxList /> renders with touched 1`] = `
<fieldset
  class="fieldset fieldset--success fieldset--touched"
  id="field_17"
>
  <legend
    class="fieldset__legend fieldset__legend--success fieldset__legend--touched"
  />
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_18"
      type="checkbox"
      value="foo"
    />
    <label
      class="field__label"
      for="field_18"
    >
      Foo
    </label>
  </div>
  <div
    class="field"
  >
    <input
      class="field__input"
      id="field_19"
      type="checkbox"
      value="bar"
    />
    <label
      class="field__label"
      for="field_19"
    >
      Bar
    </label>
  </div>
  <div
    class="field field--disabled"
  >
    <input
      class="field__input field__input--disabled"
      disabled=""
      id="field_20"
      type="checkbox"
      value="buzz"
    />
    <label
      class="field__label field__label--disabled"
      for="field_20"
    >
      Disabled
    </label>
  </div>
</fieldset>
`;
